tools/sklearn/main_macros.xml

```diff
@@ -104,6 +101,77 @@ def get_X_y(params, file1, file2):
   return X, y
   </token>
 
+  <token name="@GET_SEARCH_PARAMS_FUNCTION@">
+def get_search_params(params_builder, in_pipeline=False):
+
+    search_params = {}
+
+    for ep in params_builder['estimator_param_set']:
+        if ep['estimator_param'].strip() == '':
+            continue
+        lst = ep['estimator_param'].split(":")
+        if len(lst) >2:
+            exec lst[2].strip()
+        literal = lst[1].strip().replace("__sq__","'")\
+            .replace('__dq__','"').replace("__oc__", "{").replace("__cc__", "}")\
+            .replace("__ob__", "[").replace("__cb__", "]")
+        try:
+            ev = ast.literal_eval(literal)
+        except ValueError:
+            ev = eval(literal)
+        if in_pipeline:
+            search_params['estimator__' + lst[0].strip()] = ev
+        else:
+            search_params[lst[0].strip()] = ev
+    if not in_pipeline:
+        return search_params
+    
+    for ep in params_builder['prep_param_set']:
+        if ep['prep_param'].strip() == '':
+            continue
+        lst = ep['prep_param'].split(":")
+        if len(lst) >2:
+            exec lst[2].strip()
+        literal = lst[1].strip().replace("__sq__","'")\
```

Comment:
here as well, if you don't want to use a sanatizer, this should be a python function as you use it 3 times

Reply:
Not exactly. The first one is somewhat different.